pedsnetdcc/transform_runner.py

```diff
@@ -0,0 +1,332 @@
+import logging
+import time
+
+import sqlalchemy
+import sqlalchemy.dialects.postgresql
+
+from pedsnetdcc import TRANSFORMS, VOCAB_TABLES
+from pedsnetdcc.db import Statement, StatementSet, StatementList
+from pedsnetdcc.dict_logging import secs_since
+from pedsnetdcc.indexes import add_indexes, drop_indexes
+from pedsnetdcc.foreign_keys import add_foreign_keys, drop_foreign_keys
+from pedsnetdcc.primary_keys import add_primary_keys
+from pedsnetdcc.not_nulls import set_not_nulls
+from pedsnetdcc.schema import (create_schema_statement, create_schema,
+                               drop_schema_statement, primary_schema,
+                               schema_exists, tables_in_schema)
+from pedsnetdcc.utils import (DatabaseError, get_conn_info_dict, combine_dicts,
+                              stock_metadata, conn_str_with_search_path,
+                              pg_error, set_logged)
+
+
+logger = logging.getLogger(__name__)
+
+
+def _transform_select_sql(model_version, site, target_schema):
+    """Create SQL for `select` statement transformations.
+
+    The `search_path` only needs to contain the source schema; the target
+    schema is embedded in the SQL statements.
+
+    The returned statements are not sufficient for the transformation;
+    the `pre_transform` needs to be run beforehand.
+
+    Returns a set of tuples of (sql_string, msg), where msg is a description
+    for the operation to be carried out by the sql_string.
+
+    :param model_version: PEDSnet model version, e.g. 2.3.0
+    :param site: site name, e.g. 'stlouis'
+    :param target_schema: schema in which to create the transformed tables
+    :return: set of tuples of SQL statement strings and messages
+    :rtype: set
+    :raise: psycopg2.ProgrammingError (from the modify_select)
+    """
+    metadata = stock_metadata(model_version)
+    metadata.info['site'] = site
+    stmt_pairs = set()
+    for table_name, table in metadata.tables.items():
+        if table_name in VOCAB_TABLES:
+            continue
+
+        select_obj = sqlalchemy.select([table])
+        join_obj = table
+
+        for transform in TRANSFORMS:
+
+            select_obj, join_obj = transform.modify_select(
+                metadata,
+                table_name,
+                select_obj,
+                join_obj)
+
+        final_select_obj = select_obj.select_from(join_obj)
+
+        table_sql = str(
+            final_select_obj.compile(
+                dialect=sqlalchemy.dialects.postgresql.dialect()))
+
+        final_sql = 'CREATE UNLOGGED TABLE {0}.{1} AS {2}'.format(
```

Comment:
From reviewing the merger module, I noticed the lack of `IF EXISTS` here, which would prevent directly restarting an aborted transformation if some or all tables had been created and populated.